crates/oxc_linter/src/rules/unicorn/no_useless_undefined.rs

```diff
@@ -0,0 +1,403 @@
+                } else if let AstKind::AssignmentPattern(assign_pattern) = parent_node_kind {
+                    let left = &assign_pattern.left;
+                    let delete_span = Span::new(left.span().end, undefined_literal.span.end);
+                    return ctx.diagnostic_with_fix(
+                        no_useless_undefined_diagnostic(undefined_literal.span),
+                        |fixer| fixer.delete_range(delete_span),
+                    );
+                } else {
+                    return;
+                }
+            }
+            AstKind::CallExpression(call_expr) => {
+                if !self.check_arguments {
+                    return;
+                }
+
+                if should_ignore(&call_expr.callee) {
+                    return;
+                }
+
+                let arguments = &call_expr.arguments;
+
+                // Ignore arguments in `Function#bind()`, but not `this` argument
+                if is_function_bind_call(call_expr) && arguments.len() != 1 {
+                    return;
+                }
+                let mut undefined_args_spans = Vec::new();
+                for i in (0..arguments.len()).rev() {
+                    let arg = &arguments[i];
+                    if is_undefined(arg) {
+                        let span = arg.span();
+                        undefined_args_spans.insert(0, span);
+                    } else {
+                        break;
+                    }
+                }
+
+                if undefined_args_spans.is_empty() {
+                    return;
+                }
+                let first_undefined_span = undefined_args_spans[0];
+                let last_undefined_span = undefined_args_spans[undefined_args_spans.len() - 1];
+                let mut start = first_undefined_span.start;
+                let mut end = last_undefined_span.end;
+
+                let remaining_count = arguments.len() - undefined_args_spans.len();
+
+                if remaining_count > 0 {
+                    let previous_argument = &arguments[remaining_count - 1];
+                    start = previous_argument.span().end;
+                }
+                // If all arguments removed, and there is trailing comma, we need remove it.
+                if remaining_count == 0 {
+                    end = call_expr.span.end - 1;
+                }
+
+                let delete_span = Span::new(start, end);
+                return ctx.diagnostic_with_fix(
+                    no_useless_undefined_diagnostic_spans(undefined_args_spans),
+                    |fixer| fixer.delete_range(delete_span),
+                );
+            }
+            _ => {}
+        }
+    }
+}
+
+#[test]
+fn test() {
+    use crate::tester::Tester;
+    let options_ignore_arguments = || Some(serde_json::json!({ "checkArguments": false }));
+    let options_ignore_arrow_function_body =
+        || Some(serde_json::json!({"checkArrowFunctionBody": false}));
+    let pass = vec![
```

Comment:
Can you please add cases for member expression assignment and class property assignment? These should all pass:
```ts
const x = { a: undefined }
const y: any = {}
y.foo = undefined

class Foo () {
  public x: number | undefined = undefined }
}

function foo(x: string | undefined = undefined) { } // maybe fix into `x?: string`?
```

Reply:
I will add cases for member expression assignment and class property assignment，but`function foo(x: string | undefined = undefined) { } maybe fix into `x?: string`?` should be fix like that: `x?: string`

I try add these case on:
https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/test/no-useless-undefined.mjs

but get `?: string | undefined`

```
valid(2): function foo(x: string | undefined = undefined) { }
  Should have no errors but had 1: [
    {
      ruleId: 'rule-to-test/no-useless-undefined',
      severity: 1,
      message: 'Do not use useless `undefined`.',
      line: 1,
      column: 38,
      nodeType: 'Identifier',
      messageId: 'no-useless-undefined',
      endLine: 1,
      endColumn: 47,
      fix: { range: [Array], text: '?: string | undefined' }
    }
  ]
```